grpc/src/main/java/com/linecorp/armeria/client/grpc/GrpcClientBuilder.java

```diff
@@ -0,0 +1,497 @@
+import com.linecorp.armeria.unsafe.grpc.GrpcUnsafeBufferUtil;
+
+import io.grpc.CallCredentials;
+import io.grpc.ClientInterceptor;
+import io.grpc.Codec;
+import io.grpc.Compressor;
+import io.grpc.DecompressorRegistry;
+import io.grpc.ServiceDescriptor;
+
+/**
+ * Creates a new gRPC client that connects to the specified {@link URI} using the builder pattern.
+ * Use the factory methods in {@link GrpcClients} if you do not have many options to override.
+ */
+@UnstableApi
+public final class GrpcClientBuilder extends AbstractClientOptionsBuilder {
+
+    private final ImmutableList.Builder<ClientInterceptor> interceptors = ImmutableList.builder();
+    @Nullable
+    private final EndpointGroup endpointGroup;
+
+    @Nullable
+    private URI uri;
+    @Nullable
+    private String path;
+    private Scheme scheme;
+
+    GrpcClientBuilder(URI uri) {
+        requireNonNull(uri, "uri");
+        checkArgument(uri.getScheme() != null, "uri must have scheme: %s", uri);
+        endpointGroup = null;
+        this.uri = uri;
+        scheme = Scheme.parse(uri.getScheme());
+        validateOrSetSerializationFormat();
+    }
+
+    GrpcClientBuilder(Scheme scheme, EndpointGroup endpointGroup) {
+        requireNonNull(scheme, "scheme");
+        requireNonNull(endpointGroup, "endpointGroup");
+        uri = null;
+        this.scheme = scheme;
+        validateOrSetSerializationFormat();
+        this.endpointGroup = endpointGroup;
+    }
+
+    private void validateOrSetSerializationFormat() {
+        if (scheme.serializationFormat() == SerializationFormat.NONE) {
+            // If not set, gRPC protobuf is used as a default serialization format.
+            serializationFormat(GrpcSerializationFormats.PROTO);
+        } else {
+            ensureGrpcSerializationFormat(scheme.serializationFormat());
+        }
+    }
+
+    /**
+     * Sets the gRPC {@link SerializationFormat}. If not set, the {@link Scheme#serializationFormat()}
+     * specified when creating this builder will be used by default.
+     *
+     * @see GrpcSerializationFormats
+     */
+    public GrpcClientBuilder serializationFormat(SerializationFormat serializationFormat) {
+        requireNonNull(serializationFormat, "serializationFormat");
+        ensureGrpcSerializationFormat(serializationFormat);
+        scheme = Scheme.of(serializationFormat, scheme.sessionProtocol());
+        if (uri != null) {
+            final String rawUri = uri.toString();
+            uri = URI.create(scheme + rawUri.substring(rawUri.indexOf(':')));
+        }
+        return this;
+    }
+
+    private static void ensureGrpcSerializationFormat(SerializationFormat serializationFormat) {
+        checkArgument(GrpcSerializationFormats.isGrpc(serializationFormat),
+                      "serializationFormat: %s (expected: one of %s)",
+                      serializationFormat, GrpcSerializationFormats.values());
+    }
+
+    /**
+     * Sets the path for the endpoint.
+     */
```

Comment:
Could you add some comment on when you'd want to specify a custom path in gRPC with an example?